### PR TITLE
Add DLT/LINK_TYPE for Marvell Ethernet Switch DSA TAG

### PR DESCRIPTION
**Pull request by lunn**
Make use of the newly added infrastructure for DSA Ethernet
switches to support Marvell DSA tags.

### CODE REVIEW

**Comment by lunn** [comment]
v2: Fix author on patch

**Comment by guyharris** [comment]
Do you have a description of the switch tag, similar to the one for the [Broadcom switch tag](https://www.tcpdump.org/linktypes/broadcom-switch-tag.html)?

File: pcap-common.c [hunk 1]
```diff
@@ -1155,7 +1155,12 @@
  */
 #define LINKTYPE_IEEE802_15_4_TAP       283
 
-#define LINKTYPE_MATCHING_MAX	283		/* highest value in the "matching" range */
+/*
+ * Marvell Ethernet switches mv88e6xxx 4 byte propreitary taggging format.
```
Comment (guyharris) [inline pcap-common.c]: "proprietary", not "propreitary".

File: pcap/dlt.h [hunk 2]
```diff
@@ -1440,6 +1440,11 @@
  */
 #define DLT_IEEE802_15_4_TAP    283
 
+/*
+ * Marvell Ethernet switches mv88e6xxx 4 byte propreitary taggging format.
```
Comment (guyharris) [inline pcap/dlt.h]: Same typo for "proprietary".

**Comment by guyharris** [comment]
> Do you have a description of the switch tag, similar to the one for the [Broadcom switch tag](https://www.tcpdump.org/linktypes/broadcom-switch-tag.html)?

Or, if it's explicitly documented by Marvell, a link to Marvell's description?

**Comment by lunn** [comment]
Fixed the two spelling errors.
Add a best link i could find to Marvell Documentation in the commit message.

**Comment by guyharris** [comment]
> Add a best link i could find to Marvell Documentation in the commit message.

Is that the "Marvell Header" described in [section 8.5 "Marvell® Header Support"](https://lafibre.info/images/free/201101_Marvell_Kirkwood_88F6281_3_Functional_Spec.pdf)?

Are the additional fields mentioned in the commit message handled by the Linux code, so that the information about them is available?

**Comment by lunn** [comment]
This is the DSA header, section  8.6 Distributed Switching Architecture (DSA)
The additional fields are currently not used by the kernel, however tcpdump can dump them, so they are documented there.
Marvell have two headers, EDSA and DSA. The EDSA header is an Ether type, followed by two reserved bytes, and then a DSA header. tcpdump knows how to print an EDSA header, i added that a while back. So making it print the DSA header is just a bit of refactoring.

**Comment by lunn** [comment]
Hi Guy

What do i need to do to make progress with this patch?

Thanks
    Andrew

**Comment by guyharris** [comment]
> What do i need to do to make progress with this patch?

Write up a complete description of the header in the style of, for example, [the Broadcom switch tag](https://www.tcpdump.org/linktypes/broadcom-switch-tag.html), showing all the fields, including the "additional fields" not documented in the 88F6180/88F619x/88F6281 functional specifications.

**Comment by guyharris** [comment]
Handled by 62b2f3aea00a7bb8a066311d699c4585b78bd247.

**Pull request closed without merging**